CorsixTH/Lua/research_department.lua

```diff
@@ -505,39 +506,41 @@ function ResearchDepartment:discoverObject(object, automatic)
   self.research_progress[object].discovered = true
 
   -- Go through all rooms to see if another one can be made available.
-  for room, _ in pairs(self.hospital.undiscovered_rooms) do
-    local discovery = true
-    for needed, _ in pairs(room.objects_needed) do
-      local obj = self.research_progress[TheApp.objects[needed]]
-      if obj and not obj.discovered then
-        discovery = false
-        break
-      end
-    end
-    if discovery then
-      self.hospital.discovered_rooms[room] = true
-      self.hospital.undiscovered_rooms[room] = nil
-      if self.hospital == self.world.ui.hospital then
-        if automatic then
-          self.world.ui.adviser:say(_A.research.new_available:format(object.name))
-        else
-          self.world.ui.adviser:say(_A.research.new_machine_researched:format(object.name))
+  for i, room_id in ipairs(self.hospital.room_discovery) do
```

Comment:
It's also not a `room_id`. This one is more confusing, as the code already has `room_id`, with a very specific meaning, and you are not using the same meaning here I think.